packages/token-bridge-sdk/src/hooks/arbTokenBridge.types.ts

```diff
@@ -48,24 +48,75 @@ export enum AssetType {
 
 export type TransactionLifecycle<Tx, TxReceipt> = Partial<{
   onTxSubmit: (tx: Tx) => void
-  onTxConfirm: (txReceipt: TxReceipt) => void
+  onTxConfirm: (tx: Tx, txReceipt: TxReceipt) => void
 }>
 
-export type L1EthDepositTransactionLifecycle = TransactionLifecycle<
-  L1EthDepositTransaction,
-  L1EthDepositTransactionReceipt
->
+export type TokenTransactionLifecycle<Tx, TxReceipt> = Pick<
+  TransactionLifecycle<Tx, TxReceipt>,
+  'onTxConfirm'
+> &
+  Partial<{
+    onTxSubmit: (tx: Tx, symbol: string) => void
+  }>
 
-export type L1ContractCallTransactionLifecycle = TransactionLifecycle<
-  L1ContractCallTransaction,
-  L1ContractCallTransactionReceipt
->
+export type L1EthDepositTransactionLifecycle = Pick<
+  TransactionLifecycle<L1EthDepositTransaction, L1EthDepositTransactionReceipt>,
+  'onTxSubmit'
+> & {
+  onTxConfirm: (
+    tx: L1EthDepositTransaction,
+    txReceipt: L1EthDepositTransactionReceipt,
+    ethDepositMessage: EthDepositMessage
+  ) => void
+}
+
+export type TokenL1ContractCallTransactionLifecycle = Pick<
+  TokenTransactionLifecycle<
+    L1ContractCallTransaction,
+    L1ContractCallTransactionReceipt
+  >,
+  'onTxSubmit'
+> & {
+  onTxConfirm: (
+    tx: L1ContractCallTransaction,
+    txReceipt: L1ContractCallTransactionReceipt,
+    l1Tol2Message: IL1ToL2MessageReader
+  ) => void
+}
 
 export type L2ContractCallTransactionLifecycle = TransactionLifecycle<
   L2ContractTransaction,
   L2TransactionReceipt
 >
 
+export type TokenL2ContractCallTransactionLifecycle = TokenTransactionLifecycle<
+  L2ContractTransaction,
+  L2TransactionReceipt
+>
+
+export type TokenContractTransactionLifecycle = TokenTransactionLifecycle<
+  ContractTransaction,
+  ContractReceipt
+>
```

Comment:
I think both of these need to use `L2ContractTransaction` and `L2TransactionReceipt`? They should be the same, so we can have it be a single type called `EthOrTokenWithdrawalTransactionLifecycle`

We'll need `onL2TxSubmit`, `onL2TxSuccess` and `onL2TxFailure`

Reply:
`TokenL2ContractCallTransactionLifecycle` is used in `withdraw` (L2)

`TokenContractTransactionLifecycle` is used in `approve` (L1) and `approveL2` (LPT) 

let me see how to organise them better @_@